ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,280 @@
+    """Class to check optional input parameters"""
+
+    def check(self):
+        """Method checks if optional parameter exists and raises ValueError exception if it has unexpected type"""
+        if self.parameter is not None:
+            check_parameter_type(
+                parameter=self.parameter,
+                parameter_name=self.parameter_name,
+                expected_type=self.expected_type,
+            )
+
+
+def check_file_extension(
+    file_path: str, file_path_name: str, expected_extensions: list
+):
+    """Function raises ValueError exception if file has unexpected extension"""
+    file_extension = file_path.split(".")[-1].lower()
+    if file_extension not in expected_extensions:
+        raise ValueError(
+            f"Unexpected extension of {file_path_name} file. expected: {expected_extensions} actual: {file_extension}"
+        )
+
+
+def check_that_null_character_absents_in_string(parameter: str, parameter_name: str):
+    """Function raises ValueError exception if null character: '\0' is specified in path to file"""
+    if "\0" in parameter:
+        raise ValueError(f"\\0 is specified in {parameter_name}: {parameter}")
+
+
+def check_that_file_exists(file_path: str, file_path_name: str):
+    """Function raises ValueError exception if file not exists"""
+    if not exists(file_path):
+        raise ValueError(
+            f"File {file_path} specified in '{file_path_name}' parameter not exists"
+        )
+
+
+def check_that_parameter_is_not_empty(parameter, parameter_name):
+    """Function raises ValueError if parameter is empty"""
+    if not parameter:
+        raise ValueError(f"parameter {parameter_name} is empty")
+
+
+def check_that_all_characters_printable(parameter, parameter_name):
+    """Function raises ValueError if one of string-parameter characters is not printable"""
+    if not all(c.isprintable() for c in parameter):
+        raise ValueError(
+            f"parameter {parameter_name} has not printable symbol: {parameter}"
+        )
+
+
+class InputConfigCheck(BaseInputArgumentChecker):
+    """Class to check input config_parameters"""
+
+    def __init__(self, parameter):
+        self.parameter = parameter
+
+    def check(self):
+        """Method raises ValueError exception if "input_config" parameter is not equal to expected"""
+        parameter_name = "input_config"
+        raise_value_error_if_parameter_has_unexpected_type(
+            parameter=self.parameter,
+            parameter_name=parameter_name,
+            expected_type=(str, DictConfig, dict),
+        )
+        check_that_parameter_is_not_empty(
+            parameter=self.parameter, parameter_name=parameter_name
+        )
+        if isinstance(self.parameter, str):
+            check_that_null_character_absents_in_string(
+                parameter=self.parameter, parameter_name=parameter_name
+            )
+            if isinstance(safe_load(self.parameter), str):
```

Comment:
Please, use `yaml.safe_load` instead of just `safe_load`. Otherwise it is not obvious if it is your function or yaml.

Reply:
**Attention**
And why do you make `yaml.safe_load` and expects just one string inside it? Any arbitrary structure may be in YAML, if a YAML string is used to keep only one string it is really strange.

Reply:
thanks, reworked "check" method of InputConfigCheck class